src/backend/integrations/builtin/obs/effects/transform-obs-source-scale.ts

```diff
@@ -0,0 +1,238 @@
+                        placeholder-text="Current Rotation"
+                        model="effect.startTransform.rotation"
+                        style="flex-basis: 50%" />
+                    <firebot-input
+                        input-title="End Rotation"
+                        placeholder-text="0 - 360"
+                        model="effect.endTransform.rotation"
+                        style="flex-basis: 50%" />
+                </div>
+            </div>
+        </eos-container>
+    `,
+    optionsController: ($scope: any, backendCommunicator: any, $q: any) => {
+        $scope.isObsConfigured = false;
+
+        $scope.scenes = [];
+        $scope.sceneItems = [];
+
+        $scope.selectScene = (sceneName: string) => {
+            $scope.effect.sceneItem = undefined;
+            $scope.getSources(sceneName);
+        };
+
+        $scope.selectSceneItem = (sceneItem: OBSSceneItem) => {
+            $scope.effect.sceneItem = sceneItem;
+        };
+
+        $scope.getScenes = () => {
+            $scope.isObsConfigured = backendCommunicator.fireEventSync("obs-is-configured");
+
+            $q.when(backendCommunicator.fireEventAsync("obs-get-scene-list")).then(
+                (scenes: string[] | undefined) => {
+                    $scope.scenes = scenes?.map(scene => ({ name: scene, custom: false })) ?? [];
+                    $scope.scenes.push($scope.customScene);
+
+                    if ($scope.effect.sceneName != null) {
+                        $scope.getSources($scope.effect.sceneName);
+                    }
+                }
+            );
+        };
+        $scope.getScenes();
+
+        $scope.getSources = (sceneName: string) => {
+            $scope.isObsConfigured = backendCommunicator.fireEventSync("obs-is-configured");
+
+            $q.when(
+                backendCommunicator.fireEventAsync("obs-get-transformable-scene-items", [sceneName])
+            ).then((sceneItems: OBSSceneItem[]) => {
+                $scope.sceneItems = sceneItems ?? [];
+            });
+        };
+    },
+    optionsValidator: (effect) => {
+        if (effect.sceneName == null) {
+            return ["Please select a scene."];
+        }
+        if (effect.sceneItem == null) {
+            return ["Please select a source."];
+        }
+        if (effect.duration == null) {
+            return ["Please enter a duration."];
+        }
+        return [];
+    },
+    onTriggerEvent: async ({ effect }) => {
+        const parsedStart: Record<string, number> = {};
+        const parsedEnd: Record<string, number> = {};
+        const transformKeys = [];
+        if (effect.isTransformingPosition) {
+            transformKeys.push("positionX", "positionY");
+        }
+        if (effect.isTransformingScale) {
+            transformKeys.push("scaleX", "scaleY");
+        }
+        if (effect.isTransformingRotation) {
+            transformKeys.push("rotation");
+        }
```

Comment:
I think increased type safety on the expected transform keys would be advantageous here for maintainability as it will help prevent whoopsies from other maintainers / our future selves.

Something like:
```ts
type TransformKey = "positionX" | "positionY" | "scaleX" | "scaleY" | "rotation";
```

And then use that anywhere where those keys are expected (eg `const parsedStart: Record<TransformKey, number> = {}`, `const transformKeys: TransformKey = [];`, etc)

(might want to define and export the type in `obs-remote.ts` since it needs to be utilized there too)

Reply:
Created a Key typing in `obs-remote`